tuning/trainercontroller/operations/operation.py

```diff
@@ -0,0 +1,45 @@
+# Standard
+import abc
+import re
+import inspect
+
+class Operation(metaclass=abc.ABCMeta):
+    """Base class for operations"""
+
+    def __init__(self):
+        """Initializes the HuggingFace controls. In this init, we follow the convention that
+        every action should preceed with prefix `should_`. If so, it is treated as a valid
+        action.
+        """
+        self.valid_actions = {}       
+        for action_name, action_method in inspect.getmembers(self, predicate=inspect.ismethod):
+            if re.search(r'^should_', action_name) is not None:
```

Comment:
method name just being `should_` would also pass through isn't it? Do we want to allow that?
```suggestion
            if re.search(r'^should_.+', action_name) is not None:
```

Reply:
Addressed.